File: iOS/CurrencyConverter/CurrencyConverter/Common/Data/Models/CountryCurrenciesAndFlag.swift
//
//  CountryCurrenciesAndFlag.swift
//  CurrencyConverter
//
//  Created by Ahmad Mahmoud on 14/12/2020.
//

struct CountryCurrenciesAndFlag: Decodable {
    
    let flag: String
    var currencies: [Currency] = []
    
    func hasCurrencyWithSymbol(symbol: String) -> Bool {
        return currencies.firstIndex(where: { $0.code == symbol }) != nil
    }
}

File: Android/CurrencyConverter/app/src/main/java/com/ams/currencyconverter/common/data/models/CountryCurrenciesAndFlag.kt
package com.ams.currencyconverter.common.data.models

data class CountryCurrenciesAndFlag(val flag: String?, val currencies: List<Currency>) {
    fun hasCurrencyWithSymbol(symbol: String): Boolean {
        return currencies.find { it.code == symbol }?.let { true } ?: false
    }
}

